lib/internal/v8.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+function previewMapIterator(it, n) {
+  it = %MapIteratorClone(it);
+  const result = [];
+  const out = [,,];
+  while (n-- > 0) {
+    const rc = %MapIteratorNext(it, out);
+    if (rc === 0) break;
+    if (rc === 1) result.push(out[0]);       // it.keys()
```

Comment:
Can the 0/1/2/3 magic numbers be constants instead? V8 is using

```
define ITERATOR_KIND_KEYS = 1;
define ITERATOR_KIND_VALUES = 2;
define ITERATOR_KIND_ENTRIES = 3;
```

which looks good to me.